src/compiler/parser.ts

```diff
@@ -6647,8 +6671,61 @@ namespace ts {
                     }
                 }
 
+                function textsEqual(parent: EntityName, name: EntityName): boolean {

```

Comment:
Nit: would just name these variables `a` and `b` since they don't really depend on a parent-child context.